src/deserialize/mod.rs

```diff
@@ -177,6 +186,128 @@ where
+                .map(|(key, e)| (key, Deserialize::deserialize(e).unwrap()))
+                .fold(HashMap::new(), |mut acc, (key, e)| {
+                    acc.insert(key.to_string(), e);
+                    acc
+                })),
+            Edn::NamespacedMap(ns, _) => Ok(edn
+                .map_iter()
+                .ok_or_else(|| Error::Iter(format!("Could not create iter from {:?}", edn)))?
+                .map(|(key, e)| {
+                    (
+                        if ns.starts_with(":") {
+                            ns.to_string() + "/" + key
+                        } else {
+                            String::from(":") + ns + "/" + key
+                        },
+                        Deserialize::deserialize(e).unwrap(),
+                    )
+                })
+                .fold(HashMap::new(), |mut acc, (key, e)| {
+                    acc.insert(key.to_string(), e);
+                    acc
+                })),
+            _ => Err(build_deserialize_error(
+                &edn,
+                std::any::type_name::<HashMap<String, T>>(),
+            )),
+        }
+    }
+}
+
+impl<T> Deserialize for BTreeMap<String, T>
+where
+    T: Deserialize,
+{
+    fn deserialize(edn: &Edn) -> Result<Self, Error> {
+        match edn {
+            Edn::Map(_) => Ok(edn
+                .map_iter()
+                .ok_or_else(|| Error::Iter(format!("Could not create iter from {:?}", edn)))?
+                .map(|(key, e)| (key, Deserialize::deserialize(e).unwrap()))
+                .fold(BTreeMap::new(), |mut acc, (key, e)| {
+                    acc.insert(key.to_string(), e);
+                    acc
+                })),
+            Edn::NamespacedMap(ns, _) => Ok(edn
+                .map_iter()
+                .ok_or_else(|| Error::Iter(format!("Could not create iter from {:?}", edn)))?
+                .map(|(key, e)| {
+                    (
+                        if ns.starts_with(":") {
+                            ns.to_string() + "/" + key
+                        } else {
+                            String::from(":") + ns + "/" + key
+                        },
+                        Deserialize::deserialize(e).unwrap(),
+                    )
+                })
+                .fold(BTreeMap::new(), |mut acc, (key, e)| {
+                    acc.insert(key.to_string(), e);
+                    acc
+                })),
+            _ => Err(build_deserialize_error(
+                &edn,
+                std::any::type_name::<BTreeMap<String, T>>(),
+            )),
+        }
+    }
+}
+
+impl<T: std::cmp::Eq + std::hash::Hash> Deserialize for HashSet<T>
+where
+    T: Deserialize,
+{
+    fn deserialize(edn: &Edn) -> Result<Self, Error> {
+        match edn {
+            Edn::Set(_) => Ok(edn
+                .set_iter()
+                .ok_or_else(|| Error::Iter(format!("Could not create iter from {:?}", edn)))?
+                .map(|e| Deserialize::deserialize(e).unwrap())
```

Comment:
I think we should favor `from_edn` instead of `Deserialize::deserialize`

Reply:
Also, shouldn't we return the possible errors too? This seems unsafe

Reply:
every other deserialize uses `Deserialize::deserialize` and I think `from_edn`is something more usable from the client side, as well as `Deserialize::deserialize` is more explicit to have in an internal library context

Reply:
couldn't resolve `unwrap`